helper-cli/src/main/kotlin/commands/MergeAnalyzerResultsCommand.kt

```diff
@@ -0,0 +1,291 @@
+import java.util.LinkedList
+import java.util.SortedMap
+import java.util.SortedSet
+import java.util.TreeMap
+import java.util.TreeSet
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.VcsInfo
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.model.config.Curations
+import org.ossreviewtoolkit.model.config.Excludes
+import org.ossreviewtoolkit.model.config.LicenseChoices
+import org.ossreviewtoolkit.model.config.RepositoryConfiguration
+import org.ossreviewtoolkit.model.config.Resolutions
+import org.ossreviewtoolkit.model.readValue
+import org.ossreviewtoolkit.model.writeValue
+import org.ossreviewtoolkit.utils.common.expandTilde
+import org.ossreviewtoolkit.utils.core.Environment
+
+class MergeAnalyzerResultsCommand : CliktCommand(
+    help = "Read multiple analyzer result files and merge them into one combined analyzer result file."
+) {
+    companion object {
+        private val utcClock = Clock.systemUTC()
+        private fun now(): Instant = ZonedDateTime.now(utcClock).toInstant()
+    }
+
+    private val inputAnalyzerResultFiles by option(
+        "--input-analyzer-result-files", "-i",
+        help = "A comma separated list of analyzer result files to be merged."
+    ).convert { File(it.expandTilde()).absoluteFile.normalize() }.split(",").required()
+
+    private val outputAnalyzerResultFile by option(
+        "--output-analyzer-result-file", "-o",
+        help = "The output analyer file."
+    ).convert { it.expandTilde() }
+        .file(mustExist = false, canBeFile = true, canBeDir = false, mustBeWritable = false, mustBeReadable = false)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val inputOrtResults: MutableList<AnalyzerRun> = LinkedList()
+        val inputRepositories: MutableList<Repository> = LinkedList()
+
+        inputAnalyzerResultFiles.stream()
+            .map { it.readValue<OrtResult>() }
+            .forEach {
+                it.analyzer.let { value -> inputOrtResults.add(value!!) }
+                inputRepositories.add(it.repository)
+            }
+
+        val analyzerRun = AnalyzerRun(
+            startTime = now(),
+            endTime = now(),
+            environment = Environment(),
+            config = aggregateAnalyzerConfiguration(inputOrtResults),
+            result = aggregrateAnalyzerRun(inputOrtResults)
+        )
+
+        val repository = Repository(
+            vcs = analyzerRun.result.projects.first().vcs,
+            vcsProcessed = analyzerRun.result.projects.first().vcsProcessed,
```

Comment:
ORT requires in several places that it can find the relative path of a project to the analyzer root repository. This requirement cannot be fulfilled when merging multiple analyzer result files from independent repositories, because the model was never designed to support this. Taking the VCS of the arbitrary first project here will cause issues with the handling of license findings and repository configurations in later steps, this should at least be mentioned in the documentation.

Reply:
This limitation is well-known and therefore the merge code should assert that all merged analyzer results orignate from the same 'root' VCS.
I just realized that I have not implemented that in the first pull request

The helper command was by the need to handle the case of a multiple product teams working on individual project directories in a shared repository with each team delivering its own analyzer result.
From a logical perspective, it would not make any sense to merge analyzer results from arbitary sources.